fileNames/helpers.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Find each FileUpload node in a flow (live or published) and update its' passport variable (fn)
+ */
+const updateUploadNodeFn = (flowData) => {
+  let newFlowData = flowData;
+  Object.entries(flowData)
+    .filter(([_nodeId, nodeData]) => nodeData["type"] === 140)
+    .forEach(([uploadNodeId, uploadNodeData]) => {
+      const currentFn = uploadNodeData["data"]["fn"];
+      if (existingFns.includes(currentFn)) {
+        newFlowData[uploadNodeId]["data"]["fn"] = filesNowToNext[currentFn];
+      }
+    });
+  return newFlowData;
+}
+
+/**
+ * Find each FileUploadAndLabel node in a flow (live or published) and update the passport variable (fn) of each of its' defined files
+ */
+const updateUploadAndLabelNodeFns = (flowData) => {
+  let newFlowData = flowData;
+  Object.entries(flowData)
+    .filter(([_nodeId, nodeData]) => nodeData["type"] === 145)
+    .forEach(([uploadAndLabelNodeId, uploadAndLabelNodeData]) => {
+      uploadAndLabelNodeData["data"]["fileTypes"].forEach((file, i) => {
+        const currentFn = file?.fn;
+        if (existingFns.includes(currentFn)) {
+          newFlowData[uploadAndLabelNodeId]["data"]["fileTypes"][i]["fn"] = filesNowToNext[currentFn];
+        }
+      });
+    });
+  return newFlowData;
+}
+
+/**
+ * Find each TextInput (110) or DateInput (120) node in a flow (live or published) that asks about what a file contains and update its' passport variable (fn)
+ */
```

Comment:
See this flow as example of these "supporting inputs" for files: https://editor.planx.dev/opensystemslab/certificate-of-lawfulness-documents-immunity